docs: Update legacy commands to `sentry-cli files` (#10275)
getsentry/sentry-docs: docs/cli/releases.mdx

## Managing Release Artifacts

When you're working with JavaScript and other platforms, you can upload release artifacts (which are taken into account during processing) to Sentry. The `sentry-cli` has specific support for [source maps](/platforms/javascript/sourcemaps/), which are our most common release artifacts.

> To manage release artifacts the `sentry-cli files` command can be used which itself provides various sub commands.

Reviewer: ```suggestion
The `sentry-cli files` command (which provides various sub-commands) can be used to manage release artifacts.
```

Author: Included in #10288

### Upload Files

> The most common use case is to upload files. For the generic upload the `sentry-cli files upload` command can be used. However since most release artifacts are JavaScript source map related we have a [Upload Source Maps](#sentry-cli-sourcemaps) convenience method for that.

Reviewer: ```suggestion
The most common use case is to upload files. For the generic upload, the `sentry-cli files upload` command can be used. However, since most release artifacts are JavaScript source map related, we have an [Upload Source Maps](/cli/releases/#upload-source-maps) convenience method for that.
```

Author: Included in https://github.com/getsentry/sentry-docs/pull/10288

> When you are working with JavaScript and other platforms, you can upload release artifacts to Sentry which are then considered during processing. The most common release artifact are [source maps](/platforms/javascript/sourcemaps/) for which `sentry-cli` has specific support.

Reviewer: ```suggestion
When you're working with JavaScript and other platforms, you can upload release artifacts (which are taken into account during processing) to Sentry. The `sentry-cli` has specific support for [source maps](/platforms/javascript/sourcemaps/), which are our most common release artifacts.
```

Author: I moved this commit to #10288

> The first argument to `upload` is the path to the file, the second is an optional URL we should associate it with. Note that if you want to use an abbreviated URL (eg: `~/foo.js`) make sure to use single quotes to avoid the expansion by the shell to your home folder.

Reviewer: ```suggestion
The first argument to `upload` is the path to the file. The second is an optional URL we should associate it with. If you want to use an abbreviated URL (`~/foo.js`, for example), make sure you're using single quotes to avoid the shell expansion to your home folder.
```

Author: Included in https://github.com/getsentry/sentry-docs/pull/10288
